chromium_src/components/sync/base/model_type_unittest.cc

```diff
@@ -12,5 +12,10 @@ TEST_F(ModelTypeTest, EncryptableUserTypes) {
   EXPECT_TRUE(EncryptableUserTypes().Has(DEVICE_INFO));
 }
 
+// This test is supposed to fail when sync types are increased/decreased
+TEST_F(ModelTypeTest, ModelTypeCounts) {
+  EXPECT_EQ(ModelType::NUM_ENTRIES, 41);
```

Comment:
Is a situation possible when types are removed and added so the total doesn't change, but the types change?

Reply:
Also, it seems if EncryptableUserTypes() is modified this wouldn't alert us. Do we want to maybe have an explicit set of types we know are returned by EncryptableUserTypes() now and compare it to the result of the EncryptableUserTypes() call?

Reply:
>Is a situation possible when types are removed and added so the total doesn't change, but the types change?

Yes, I will use `ModelTypeForHistograms::kMaxValue` instead because it can't be renumbered.

>Also, it seems if EncryptableUserTypes() is modified this wouldn't alert us. Do we want to maybe have an explicit set of types we know are returned by EncryptableUserTypes() now and compare it to the result of the EncryptableUserTypes() call?

`EncryptableUserTypes()` is just a subset of all types, why would it be alerted?

Reply:
Oh I see, didn't look at the original thread carefully - thought we only cared about encryptable types.

Reply:
fixed https://github.com/brave/brave-core/pull/5923/commits/a8f026629b5b3639d5facc9891815b86ea40f24e